autopilot/interface.go

```diff
@@ -90,6 +90,9 @@ type AttachmentDirective struct {
 	// a ChannelOpener implementation to execute the directive.
 	PeerKey *btcec.PublicKey
 
+	// NodeID is the serialized compressed pubkey of the target node.
+	NodeID NodeID
```

Comment:
`PeerKey` still needed?

Reply:
yep, we pass it to `ConnectToPeer`

Reply:
Is it the same key though? Maybe it would make sense to call it `PeerKeyBytes` to clarify.

Reply:
Yeah they’re the same key. That’s what it was previously, though our only use of this field was to cast `PeerKeyBytes` into `NodeID`s within the agent, so this method resulted in less code duplication.

I agree the the `Peer`/`Node` nomenclature could be more unified, though maybe it’s worth using `NodeKey` and `NodeID` instead? This would be less verbose than `PeerKeyBytes` and also avoid casting

Reply:
SGTM 👍

Reply:
switched `PeerKey` -> `NodeKey`, reads well :)